test/Test/Stan/Cli.hs

```diff
@@ -0,0 +1,45 @@
+module Test.Stan.Cli
+    ( cliSpec
+    ) where
+
+import Options.Applicative (ParserResult (Success), execParserPure)
+import Test.Hspec (Spec, describe, it, shouldBe)
+import Trial (withTag)
+
+import Stan.Cli (StanArgs (..), StanCommand (..), stanCliParser, stanParserPrefs)
+import Stan.Config (Check (..), CheckFilter (..), CheckScope (..), CheckType (..), Config,
+                    ConfigP (..), PartialConfig, configToCliCommand)
+import Stan.Core.Id (Id (..))
+
+
+cliSpec :: Spec
+cliSpec = describe "CLI configuration tests" $ do
+    it "Converts Config to CLI command and parses it back" $
+        case execParserPure stanParserPrefs stanCliParser stanCommand of
+            Success (Stan StanArgs{..}) -> stanArgsConfig `shouldBe` partialConfigExample
+            _                           -> fail "Toml-To-CLI doesn't match the original config"
```

Comment:
Wow, this is a very nice test! I guess we can also implement a property test for this: generate random Config, convert to Text and parse and get the same config.